server/zanata-war/src/test/java/org/zanata/tmx/TMXParserTest.java

```diff
@@ -208,6 +212,14 @@ public void parseTMX() throws Exception {
         for (TransMemoryUnit tu : tm.getTranslationUnits()) {
             assertThat(tu.getTransUnitVariants().size(), greaterThan(0));
         }
+
+        Optional<String> seg = tm.getTranslationUnits().stream()
+                .filter(tu -> tu.getTransUnitId() != null && tu.getTransUnitId().equals("tuid2"))
+                .map(TransMemoryUnit::getTransUnitVariants)
+                .map(map ->  map.get("ja"))
+                .map(TransMemoryUnitVariant::getPlainTextSegment)
+                .findAny();
+        assertThat(seg.orElseGet(null), containsString("を選択します"));
```

Comment:
It didn't detect it in this case, but I still think it's a useful test.

If we were to run our unit tests in a POSIX/ASCII environment (and perhaps we should), I think it might actually have detected the problem.